app/src/main/java/com/example/smartshopper/CreateAccountActivity.java

```diff
@@ -1,13 +1,92 @@
+  PlatformHelpers platformHelpers;
+  Context context;
+  LocalStorage localStorage;
+  NavigationMenuItemView signinButton;
+  CheckBox agreeToTerms;
+
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_create_account);
+    createAccountButton = findViewById(R.id.createAccountButton);
+    usernameET = findViewById(R.id.editTextCreateUsername);
+    emailET = findViewById(R.id.editTextCreateEmail);
+    passwordET = findViewById(R.id.editTextCreatePassword);
+    confirmPasswordET = findViewById(R.id.editTextConfirmPassword);
+    signinButton = findViewById(R.id.nav_logout);
+    context = this;
+    platformHelpers = new PlatformHelpers(context);
+    localStorage = new LocalStorage(context);
+    agreeToTerms = findViewById(R.id.agreeToTermsCheckbox);
+
+    createAccountButton.setOnClickListener(view -> {
+      String username = usernameET.getText().toString();
+      String email = emailET.getText().toString();
+      String password = passwordET.getText().toString();
+      String confirmPassword = confirmPasswordET.getText().toString();
+      Log.v("Checkbox value", String.valueOf(agreeToTerms.isChecked()));
+
+//        todo: would be nice if we checked for valid email with regex here
+      if (username.equals("") || email.equals("") || password.equals("") || confirmPassword.equals("") || !agreeToTerms.isChecked()) {
+        Toast.makeText(getApplicationContext(), "All fields are required. Please verify you have filled out all fields and resubmit.", Toast.LENGTH_LONG).show();
+      }
+      else if (!password.equals(confirmPassword)) {
+        Toast.makeText(getApplicationContext(), "Password fields must match", Toast.LENGTH_LONG).show();
+      }
+      else {
+        createAccount(username, email, password);
+      }
+    });
+  }
+
+  public void createAccount(String username, String email, String password) {
+    try {
+      platformHelpers.createAccount(username, email, password, userInterface -> {
+        if (userInterface != null) {
+//          TODO: this should lead to a follow tags activity that hasn't been created yet
+          Toast.makeText(context, "Welcome to Smart Shopper "+ username + "!", Toast.LENGTH_LONG).show();
+          signIn(userInterface);
+          Intent mainActivityIntent = new Intent(context, MainActivity.class);
+          context.startActivity(mainActivityIntent);
+        }
+        else {
+          Toast.makeText(context, "Username or email taken. Please try again.", Toast.LENGTH_LONG).show();
+        }
+      });
+    } catch (Exception error) {
+      Log.v("CreateAccountActivity Error", error.toString());
+    }
+  }
+  @SuppressLint({"RestrictedApi"})
+  public void signIn(User user) {
+    try {
+      localStorage.setUser(user);
+      signinButton.setTitle("Sign Out");
+      Toast.makeText(getApplicationContext(), "Successful Login", Toast.LENGTH_LONG).show();
```

Comment:
Not really necessary to do it now, but I noticed that you're directing the user back to the MainActivity on account creation. I think it'd be a good idea to redirect back to main after sign in too

Reply:
It actually should be. And it should show a toast with a welcome and the username.

Reply:
You did! Disregard whatever I said